recipes/cpp_project_framework/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if self.settings.os != "Linux" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration(f"{self.name} is just supported for Linux and Windows")
+
+        compiler = self.settings.compiler
+
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if compiler == "gcc" or compiler == "clang":
+            if compiler.get_safe("libcxx") != "libstdc++":
+                raise ConanInvalidConfiguration(f"only supported {compiler} with libstdc++")
+
+        min_version = self._minimum_compilers_version.get(str(compiler))
+        if not min_version:
+            self.output.warn(f"{self.name} recipe lacks information about the {compiler} compiler support.")
+        else:
+            if tools.Version(compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support. The current compiler {compiler} {compiler.version} does not support it.")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        cmake = CMake(self)
+        self.run("conan install conanfile.txt -b missing -s build_type=%s -if ." % cmake.build_type, cwd=self._source_subfolder)
+        cmake.configure(source_folder=self._source_subfolder, build_folder=os.path.join(self._source_subfolder, cmake.build_type))
+        cmake.build()
+
+        # Explicit way:
+        # self.run('cmake %s/cpp_project_framework %s'
+        #          % (self.source_folder, cmake.command_line))
+        # self.run("cmake --build . %s" % cmake.build_config)
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("*.h", dst="include/%s" % self.name, src=os.path.join(self._source_subfolder, self.name))
+        self.copy("*.hpp", dst="include/%s" % self.name, src=os.path.join(self._source_subfolder, self.name))
+        self.copy("*.hxx", dst="include/%s" % self.name, src=os.path.join(self._source_subfolder, self.name))
+        for resource in self.exports_resources:
+            self.copy(resource, dst="res", src=self._source_subfolder)
+
+    def package_info(self):
```

Comment:
again if its a framework i'd expect there to be no settings in package id or package info

Reply:
The package info section is actually no need and I removed it in b99dee54c00499301e1a193101533ea23bd2bd06.  Thanks~